lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -655,9 +662,28 @@ impl ShardReplicaSet {
 
     /// Check if the are any locally disabled peers
     /// And if so, report them to the consensus
-    pub async fn sync_local_state(&self) -> CollectionResult<()> {
-        for failed_peer in self.locally_disabled_peers.read().iter() {
-            self.notify_peer_failure(*failed_peer);
+    pub async fn sync_local_state<F>(&self, get_shard_transfer: F) -> CollectionResult<()>
+    where
+        F: Fn(ShardTransferKey) -> Option<ShardTransfer>,
+    {
+        for &failed_peer in self.locally_disabled_peers.read().iter() {
+            self.notify_peer_failure(failed_peer);
+
```

Comment:
Maybe it's worth to add a comment here:

```rust
// Abort any transfers from a failed peer to prevent spamming `SetShardReplicaState(Dead)`
```

or something similar.